authz/audit/audit_logging_test.go

```diff
@@ -275,23 +279,26 @@ func (s) TestAuditLogger(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 			defer cancel()
 
-			client.UnaryCall(ctx, &testpb.SimpleRequest{})
-			client.UnaryCall(ctx, &testpb.SimpleRequest{})
+			_, err = client.UnaryCall(ctx, &testpb.SimpleRequest{})
+			validateCallResult(t, err)
```

Comment:
Inline this please, since it's so simple:

```go
if _, err := client.UnaryCall(ctx, &testpb.SimpleRequest{}); err != nil {
	t.Fatalf("Unexpected error from UnaryCall: %v", err)
}
```

Reply:
I also need to check if the err is a `permission denied` one - it's a valid scenario for the test (we count `deny` as well).
I think I can rely on `code` only but still it'll be 2 LOCs

```go
_, err = client.UnaryCall(ctx, &testpb.SimpleRequest{})
if errCode := status.Code(err); errCode != permissionDeniedStatus.Code() && errCode != codes.OK {
	t.Errorf("Call failed:%v", err)
}
```
Or I can create a helper function which takes `SimpleResponse, error` and returns back error code - smth like
```go
if errCode := errorCodeExtractor(client.UnaryCall(ctx, &testpb.SimpleRequest{})); errCode != permissionDeniedStatus.Code() && errCode != codes.OK {
	t.Errorf("Call failed:%v", err)
}
```
WDYT?

Reply:
So this is more of a problem of your test design.. You're using a table driven test but things are sufficiently different between iterations that it leads to these sorts of problems.

If you want to not care which one should happen:

```go
if _, err := UnaryCall(); err != nil && status.Code(err) != codes.PermissionDenied {
	t.Fatalf()
}
```

Or if you want to require no error when unary should not be denied and vice-versa:

```go
_, err := UnaryCall()
if test.wantAuthzOutcomes[false] > 1 /* LOL */ && status.Code(err) != codes.PermissionDenied {
	t.Fatalf()
} else if test.wantAuthzOutcomes[false] < 2 /* also LOL */ && err != nil {
	t.Fatalf()
}
```

Realistically, the latter is how your code should behave, but it's ugly.  Maybe the test should be reworked a bit or split into multiple tests, as this seems wrong.

Reply:
> My idea is to have a set of authz policies with various audit_condition values. In order to test each audit_condition I need to make a few calls and then check if audit logger was invoked as expected. That's why I think that iterations are common enough to use a table driven approach. From my perspective

Maybe add a `codes.Code` for the expected result from the Unary and Streaming RPC calls to the table?

Reply:
Agreed, I think it makes the test more readable. PTAL